packages/fury-adapter-oas3-parser/lib/parser/oas/parseServerObject.js

```diff
@@ -0,0 +1,49 @@
+const R = require('ramda');
+const { createWarning } = require('../../elements');
+const {
+  createInvalidMemberWarning, createUnsupportedMemberWarning,
+} = require('../annotations');
+const {
+  isObject, hasKey, isExtension,
+} = require('../../predicates');
+const parseObject = require('../parseObject');
+const parseString = require('../parseString');
+const pipeParseResult = require('../../pipeParseResult');
+
+const name = 'Server Object';
+const requiredKeys = ['url'];
+
+const parseMember = context => R.cond([
+  [hasKey('description'), parseString(context, name, false)],
+  [hasKey('url'), parseString(context, name, true)],
+  // [hasKey('variables'), R.compose(parseServerVariablesObject(context), getValue)], // NOT SUPPORTED YET
+  [hasKey('variables'), createUnsupportedMemberWarning(context.namespace, name)],
+  [isExtension, () => new context.namespace.elements.ParseResult()],
+  [R.T, createInvalidMemberWarning(context.namespace, name)],
+]);
+
+/**
+ * Parse the OpenAPI 'Server Object' (`#/server`)
+ * @see http://spec.openapis.org/oas/v3.0.3#server-object
+ * @returns ParseResult<Link>
+ * @private
+ */
+const parseServerObject = context => pipeParseResult(context.namespace,
+  R.unless(isObject, createWarning(context.namespace, `'${name}' is not an object`)),
+  parseObject(context, name, parseMember(context), requiredKeys, [], true),
+  (object) => {
+    const resource = new context.namespace.elements.Resource();
+
+    resource.classes.push('host');
+
+    if (object.hasKey('description')) {
+      const description = object.getValue('description');
+      resource.description = description;
```

Comment:
I forgot to update this and the below comment with the explanation so here goes.

`getValue` property on an `ObjectElement`/`ArrayElement` will return a primitive value, in this case a JavaScript String.

`get` will return the value as an element, which in this case is a `StringElement`.

Effectively, `getValue` is implemented like following:

```js
function getValue(indexOrKey) {
  const item = this.get(indexOrKey);

  if (item) {
    return item.toValue();
  }

  return undefined;
}
```

We want to set resource's description and href using the `StringElement`, because it can contain source map information which is lossed when an element is converted to a string.

Reply:
I see the point, changed